lib/active_fedora/nested_attributes.rb

```diff
@@ -148,13 +149,14 @@ def assign_nested_attributes_for_collection_association(association_name, attrib
         attributes = attributes.with_indifferent_access
 
         if attributes['id'].blank?
-          association.build(attributes.except(*UNASSIGNABLE_KEYS)) unless call_reject_if(association_name, attributes)
-
+          if !has_destroy_flag?(attributes) || !allow_destroy
```

Comment:
Would you please refactor this to `unless has_destroy_flag?(attributes) && allow_destroy`